### PR TITLE
anki: build with Python 3.9

### PR DESCRIPTION
**Pull request by kmein**
###### Description of changes

Anki 2.1.15 (the current version) is [broken](https://forums.ankiweb.net/t/error-closing-with-python-3-10-py-ssize-t-clean/14380/5) with Python 3.10.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Mindavi** [comment]
I feel like the python maintainers sometimes use `disabled = pythonNewer "3.10" ` or sometimes like that. Consider consulting the manual about that.

**Comment by smancill** [comment]
It should be

``` nix
disabled = pythonAtLeast "3.10";
```

**Comment by smancill** [comment]
It has to be an attribute of the derivation itself, not in meta.

See: https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/python-modules/boto/default.nix

**Comment by bcdarwin** [comment]
Please also consider setting `anki = python39Packages.callPackage ...` in `pkgs/top-level/all-packages.nix` for now.

File: pkgs/games/anki/default.nix [hunk 1]
```diff
@@ -192,6 +192,8 @@ buildPythonApplication rec {
     inherit manual;
   };
 
+  disabled = pythonAtLeast "3.10";
```
Comment (SuperSandro2000) [inline pkgs/games/anki/default.nix:195]: Please move this after version

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please drop the two accidental included commits.

**Comment by smancill** [comment]
I've overridden this.

I noticed that `ankisyncd` depends on `anki`, so both should use the same Python version.

File: pkgs/games/anki/default.nix (hunk 1 above)
Comment (smancill) [reply pkgs/games/anki/default.nix]: Removed this as it is unnecessary.

**Comment by smancill** [comment]
@ofborg build ankisyncd

**Pull request merged**